Map/RoomBase.gd

```diff
@@ -4,12 +4,26 @@ class_name RoomBase
 ##
 ## Holds basic functionality for a room on the map
 
-@export var light_level: int = 0
+@export var light_level: Enums.LightLevel = Enums.LightLevel.UNLIT
 @export var has_torch: bool = false
 @export var room_event: EventBase
 
+signal on_light_level_changed
+
 func _to_string() -> String:
 	return "RoomBase"
 
 func get_room_abbreviation() -> String:
 	return room_event.get_room_abbreviation()
+
+func set_torch_active() -> void:
+	has_torch = true
+	if light_level == Enums.LightLevel.DIMLY_LIT or light_level == Enums.LightLevel.LIT:
```

Comment:
this is wrong
an area that is dimly lit means the player is standing next to it and there are no torches in the range
a room that is dimly lit becomes lit if a torch is placed

I would have set the new light level by just checking the current light level and adding 1 to take the next element of the enum, with the exception of being at the max brightness where it does nothing

Reply:
So to summarize:

NOT_LIT: 

1. No torch light in range, No player adjacent

DIMLY_LIGHT: 

1. Player is in the Room, but no torch
2. Adjacent rooms that the Player is in if there are no torches placed
3. Non-Adjacent rooms that the Player is in, but is in range of torch light

LIT: 

1. Player is in the room AND torch has been placed down
2. Any other room that is DIMLY_LIT and falls within the torch range

BRIGHTLY_LIT:

1. Rooms that are LIT and falls within another torch's range

Reply:
DIMLY_LIT:
1. Player is in the room but no torch
2. Room is next to the player to player and there are no torches reaching this room
3. I'm not sure to understand but i don't see another case where it's dimly lit except the two previous cases

LIT:
1. Player doesn't need to be in the room for the room to be lit
2. Correct
3. You forgot the case of rooms in darkness that are not lit. If they are within torch range, they become lit

To sum up:
![level_of_light_explanation](https://github.com/Saplings-Projects/1M_sub/assets/49365680/6bc4f0cc-a084-440e-a193-4e5d096a6330)

Reply:
Ok thank you for the flow chart, I'm gonna add this to the documentation_images folder.